File: all/796_rotate_string/sol_circular_buffer.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        s_length = len(s)

        if s_length != len(goal):
            return False

        for start_index in range(s_length):
            for i in range(s_length):
                if (goal[i]) != (s[(i + start_index) % s_length]):
                    break
            else:
                return True

        return False


if '__main__' == __name__:
    sol = Solution()
    testcases_list = [
        ['abcde', 'cdeab'],
        ['abcde', 'abced'],
    ]

    for testcase in testcases_list:
        print(sol.rotateString(*testcase))

File: all/796_rotate_string/sol_circular_buffer.c
#include <stdbool.h>
#include <string.h>

bool rotateString(char* s, char* goal) {
    size_t s_length = strlen(s);
    size_t goal_length = strlen(goal);
    bool is_matched = false;

    if (s_length != goal_length) {
        return false;
    }

    for (size_t first_index = 0; first_index < s_length; ++first_index) {
        is_matched = true;

        for (int i = 0; i < s_length; ++i) {
            if ((goal[i]) != (s[(first_index + i) % s_length])) {
                is_matched = false;

                break;
            }
        }

        if (is_matched) {
            return true;
        }
    }

    return false;
}
